packages/events-example/src/commands/reconcile/index.ts

```diff
@@ -1,188 +1,136 @@
+    this.synchronizeDataPlane(instances, stacks);
+    this.listenForNileEvents(flags.entity, this.findLastInstance(Object.values(instances)));
   }
 
-  async run(): Promise<any> {
-    const { flags } = await this.parse(Reconcile);
-
-    // pulumi setup
-    this.localWorkspace = await LocalWorkspace.create({
-      projectSettings: { name: flags.workspace, runtime: 'nodejs' },
-    });
-    this.localWorkspace.installPlugin('aws', 'v4.0.0');
-
-    // nile setup
-    this.nile = Nile({ basePath: flags.basePath, workspace: flags.workspace });
+  async connectNile(parsedFlags: {[name: string]: any}) {
+    this.nile = Nile({ basePath: parsedFlags.basePath, workspace: parsedFlags.workspace });
     const token = await this.nile.developers
       .loginDeveloper({
         loginInfo: {
-          email: flags.email,
-          password: flags.password,
+          email: parsedFlags.email,
+          password: parsedFlags.password,
         },
       })
       .catch((error: unknown) => {
         // eslint-disable-next-line no-console
-        console.error(error);
+        console.error("Nile authentication failed", error);
       });
     this.nile.authToken = token?.token;
+  }
 
-    // load our data
-    const stacks = await this.loadPulumiStacks();
-    const instances = await this.loadInstances(
-      flags.organization,
-      flags.entity
-    );
-    let seq = 0;
-    for (const instance of Object.values(instances)) {
-      if (seq == null || (instance?.seq || seq) > seq) {
-        seq = instance?.seq || seq;
-      }
-    }
+  async loadNileInstances(organization: string, entity: string): Promise<{ [key: string]: Instance }> {
+    const instances = (
+      await this.nile.entities.listInstances({
+        org: organization,
+        type: entity,
+      })
+    )
+      .filter((value: Instance) => value !== null && value !== undefined)
+      .reduce((acc, instance: Instance) => {
+        acc[instance.id] = instance;
+        return acc;
+      }, {} as { [key: string]: Instance });
+    this.debug("Nile Instances", instances);
+    return instances;
+  }
 
-    if (flags.status) {
-      return { stacks, instances };
-    }
+  private findLastInstance(instances: Instance[]): number {
+    return instances
+      .map((value: Instance) => value?.seq || 0)
+      .reduce((prev: number, curr: number) => {
+        return Math.max(prev, curr || 0);
+      }, 0);
+  }
 
+  private synchronizeDataPlane(
+    instances: {[key: string]: Instance},
+    stacks: {[key: string]: StackSummary}
+  ) {
     // destroy any stacks that shouldnt exist
     for (const id of Object.keys(stacks)) {
       if (!instances[id]) {
-        this.destroyStack(id);
+        this.deployment.destroyStack(id);
       }
     }
 
     // create any stacks that should exist
     for (const id of Object.keys(instances)) {
       if (!stacks[id]) {
-        this.createStack(instances[id]);
+        this.deployment.createStack(instances[id]);
       }
     }
```

Comment:
@norwood - I'm seeing some inconsistencies with instances that get created then deleted before running this script. Still debugging synchronization issues.